src/operators/publish.ts

```diff
@@ -20,7 +21,7 @@ export function publish<T>(selector: MonoTypeOperatorFunction<T>): MonoTypeOpera
  * @method publish
  * @owner Observable
  */
-export function publish<T>(selector?: MonoTypeOperatorFunction<T>): MonoTypeOperatorFunction<T> {
+export function publish<T, R>(selector?: any): any {
```

Comment:
This should be `publish<T, R>(selector?: OperatorFunction<T, R>): OperatorFunction<T, R>`

Reply:
Using `OperatorFunction<T, R>` instead of `any` effects this error:

```
25   return selector ?
            ~~~~~~~~~~
26     multicast(() => new Subject<T>(), selector) :
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
27     multicast(new Subject<T>());
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

dist/src/operators/publish.ts(25,10): error TS2322: Type '((source: Observable<T>) => Observable<R>) | ((source: Observable<T>) => Observable<T>)' is not assignable to type '(source: Observable<T>) => Observable<R>'.
  Type '(source: Observable<T>) => Observable<T>' is not assignable to type '(source: Observable<T>) => Observable<R>'.
    Type 'Observable<T>' is not assignable to type 'Observable<R>'.
      Type 'T' is not assignable to type 'R'.
```

The problem is the returning of `multicast(new Subject<T>())` - that call sees the return type inferred as `OperatorFunction<T, T>` - hence the error.

The problem can be fix by using `any` here, instead of in the signature:

```
  return selector ?
    multicast(() => new Subject<T>(), selector) :
    multicast(new Subject<T>()) as any; // <-- here
```

Would you prefer that?